### PR TITLE
Added product listing page with filters

### PR DESCRIPTION
**Pull request by Shah-Arsalan**
Added product listing page 
## Filters included : 
1. Sort by category 
2. Sort by rating 
3. Sort by price ( low to high or high to low ) 
4. Sort by price ( using range ) 

## Go to FilterFunctions>FilterFunction.js for filter logic 
Product listing page is composed of three components  : ProductListAside , ProductListMain , ProductListCard ( all can be found in components folder ) 
All the 3 components are put together in productlisting.jsx inside folder WebPages

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -4,14 +4,19 @@ import Mockman from "mockman-js";
 import { Routes, Route } from "react-router-dom";
 import { Nav } from "./Components/Nav/Nav.jsx";
 import { LandingPage } from "./WebPages/LandingPage/LandingPage";
+import { ProductListing } from "./WebPages/ProductListing/ProductListing";
+import { useData } from "./Contexts/DataContext";
 
 function App() {
+  const { state } = useData();
+  console.log("the state is ", state);
```
Comment (Rohanmond) [inline src/App.js]: remove console. log

File: src/FilterFunctions/FilterFunction.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+};
+
+const sortByCategory = (state, data) => {
+  let newData = [];
+  let flag = false;
+  if (state.filters.category.tent === true) {
+    flag = true;
+    newData = union(
+      newData,
+      [...data].filter((ele) => ele.categoryName === "tent") // a,b,c
```
Comment (Rohanmond) [inline src/FilterFunctions/FilterFunction.js]: what is a,b,c

**Review by Rohanmond — APPROVED** [review]
like the folder structure, better to use variables in place of strings

File: src/Components/ProductCard/ProductCard.css [hunk 3]
```diff
@@ -0,0 +1,50 @@
+@import url(https://wanderer-store.netlify.app/main.css);
+@import url(https://wanderer-store.netlify.app//index.css);
+
+.card-skeleton .fas {
+  padding: 0.5rem;
+  background-color: rgb(238, 242, 247);
+  color: red;
```
Comment (anuj-kumary) [inline src/Components/ProductCard/ProductCard.css:7]: Don't give hardcoded color use variable

File: src/Components/ProductCard/ProductCard.jsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+const ProductCard = ({
+  ele: { _id, title, discription, price, categoryName, img, rating },
+}) => {
+  return (
+    <>
+      <div className="vertical-card-component">
+        <div className="card-skeleton">
+          <div className="imageContainer position-relative">
+            <img className="image" src={img} alt="" />
```
Comment (anuj-kumary) [inline src/Components/ProductCard/ProductCard.jsx]: Always give some value to **alt** attributes.

**Review by anuj-kumary — APPROVED** [review]
Very Clean code and UI also well structured your folder.

File: src/Components/ProductCard/ProductCard.jsx (hunk 4 above)
Comment (shrey27) [inline src/Components/ProductCard/ProductCard.jsx]: mention the alt value

**Review by shrey27 — APPROVED** [review]
Overall code structure is decent.
PR is ready to merge

**Review by mdarif6 — APPROVED** [review]
-give value to alt=""
-can remove console.log
-rest structure is okay

**Pull request merged**